context.ts

```diff
@@ -1,7 +1,55 @@
 import { createContext } from 'react';
 import moment from 'moment';
-import { InititalState } from 'typings';
-import { AVAILABILITY } from './constants';
+import { InititalState, Filters, Keycapset } from 'typings';
+
+import { AVAILABILITY, TABS } from './constants';
+import { getDayDifference } from './components/StatusLabel';
+import {
+    INTEREST_CHECK,
+    WAITING_FOR_GROUPBUY,
+    IN_GROUP_BUY,
+    ENDED,
+} from './constants';
+
+function filterByAvailability(
+    set: Keycapset,
+    availabilityFilter: string
+): boolean {
+    if (availabilityFilter === 'none') {
+        return true;
+    }
+    const { groupbuyEndDate, groupbuyStartDate, isInterestCheck } = set;
+
+    switch (availabilityFilter) {
+        case INTEREST_CHECK:
+            return isInterestCheck;
+        case WAITING_FOR_GROUPBUY:
+            return (
+                moment().diff(groupbuyStartDate, 'days') < 0 && !isInterestCheck
+            );
+        case IN_GROUP_BUY:
+            return getDayDifference(groupbuyEndDate) > 0 && !isInterestCheck;
+        case ENDED:
+            return (
+                !isInterestCheck &&
+                !(getDayDifference(groupbuyEndDate) > 0) &&
+                !(moment().diff(groupbuyStartDate, 'days') < 0)
+            );
+        default:
+            return false;
+    }
+}
+
+function handleFilters(keycapset: Keycapset, filters: Filters): boolean {
+    if (filters.activeTab === 'all' && filters.availabilityFilter === 'none')
+        return true;
+    if (filters.activeTab === 'all')
+        return filterByAvailability(keycapset, filters.availabilityFilter);
+    return (
+        keycapset.type === filters.activeTab &&
+        filterByAvailability(keycapset, filters.availabilityFilter)
+    );
+}
```

Comment:
Maybe it's nicer to put this in an 'util' directory so we make it future proof to add more filters?

Reply:
these can be moved. However this isnt a utility thats being used anywhere else, is very specific to the store filter functionality. 

some parts of it are duplicated in how the sets are given labels (this could also be added to the `keyset` object as it's added into the store, and then we wouldnt need this more complex logic to filter.... hmmmm 🤔

Reply:
there are also scenarios where we want different filtering behavior (`||` vs `&&`) and having this in the store will become more and more necessary.

Reply:
Do you mean you want to add this logic to the keycapset objects while they are put in the store after fetching? 
Could be a nice way to filter!